tests/test_example/requirements.txt

```diff
@@ -0,0 +1,47 @@
+aioredis==1.3.0
+asgiref==3.2.3
+async-generator==1.10
+async-timeout==3.0.1
+atomicwrites==1.3.0
+attrs==19.3.0
+autobahn==19.11.1
+Automat==0.8.0
+cffi==1.13.2
+# -e git+git@github.com:shjohnson-pi/channels.git@034b043567d651d49c06c7b2b7ed1f8e6823dd68#egg=channels
+channels-redis==2.4.1
+constantly==15.1.0
+coverage==4.5.4
+cryptography==2.8
+# -e git+git@github.com:shjohnson-pi/daphne.git@7d5e2c3992e42d4ff0b6a1ad93da72e7009ad60d#egg=daphne
+Django==2.2.7
+hiredis==1.0.0
+hyperlink==19.0.0
+hypothesis==4.23.0
+idna==2.8
+importlib-metadata==0.23
+incremental==17.5.0
+more-itertools==7.2.0
+msgpack==0.6.2
+packaging==19.2
+pluggy==0.13.0
+py==1.8.0
+pyasn1==0.4.7
+pyasn1-modules==0.2.7
+pycparser==2.19
+PyHamcrest==1.9.0
+pyOpenSSL==19.0.0
+pyparsing==2.4.4
+pytest==4.6.6
+pytest-asyncio==0.10.0
+pytest-django==3.6.0
+pytz==2019.3
+selenium==3.141.0
+service-identity==18.1.0
+six==1.13.0
+sqlparse==0.3.0
+Twisted==19.7.0
+txaio==18.8.1
+urllib3==1.25.6
+wcwidth==0.1.7
+zipp==0.6.0
+zope.interface==4.6.0
```

Comment:
I'm guessing that you used `pip freeze` here. My question here are all these dependencies necessary?

Reply:
That was made from a blank virtual environment, then only pip installing what was required to run the example. Then I did a pip freeze to grab all dependencies. This way if two users are comparing example output the only difference should be machine architecture.

Top dependencies are:
- channels-redis
- Django 2.2.7 (I think it fails on Django 3)
- Selenium
- Default dependencies for Daphne and Channels [tests] package.

I added the example for carltongibson, and wasn't sure if he wanted it as a .tar.gz or committed in the repo like this. My Daphne pull request https://github.com/django/daphne/pull/284 goes in more detail about this example.

I hope this answers your question.